pydantic/errors.py

```diff
@@ -324,6 +324,14 @@ def __init__(self, *, expected_arbitrary_type: AnyType) -> None:
         super().__init__(expected_arbitrary_type=display_as_type(expected_arbitrary_type))
 
 
+class ArbitraryClassError(PydanticTypeError):
```

Comment:
hmm yes I saw you call it get_restricted_type before, but I did not quite understand why you are suggesting that name.

what is restricted (or required) type supposed to mean in this context?

Reply:
mypy calls a class related to some of this `TypeType`

=> this could be called TypeTypeError, make_type_type_validator...
I just don't like _get_type_type so much (since we are getting the class not the type type)

Reply:
well, this is a "type" rather than an instance of a type, but obviously using "type error" would be massively confusing.

Using `Type[Foobar]` is saying "this must be a type/class, and further more it must be a subclass of `Foobar`", hence why I was thinking of restricted.

Why don't we use:
* `ClassError`
* `'subclass of {superclass} expected'`
* `make_class_validator`
* etc.

?

"class" makes more sense and is more specific than "type" to most people I think.

Reply:
yup that sounds good!